comp/core/workloadmeta/collectors/internal/docker/image_sbom_trivy.go

```diff
@@ -86,61 +92,65 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 	}()
 
 	go func() {
-		for result := range resultChan {
-			if result.ImgMeta == nil {
-				log.Errorf("Scan result does not hold the image identifier. Error: %s", result.Error)
-				continue
-			}
-			status := workloadmeta.Success
-			reportedError := ""
-			var report *cyclonedx.BOM
-			if result.Error != nil {
-				// TODO: add a retry mechanism for retryable errors
-				log.Errorf("Failed to generate SBOM for docker: %s", result.Error)
-				status = workloadmeta.Failed
-				reportedError = result.Error.Error()
-			} else {
-				bom, err := result.Report.ToCycloneDX()
-				if err != nil {
-					log.Errorf("Failed to extract SBOM from report")
+		for {
+			select {
+			case result, ok := <-resultChan:
+				if !ok {
+					return
+				}
+				if result.ImgMeta == nil {
+					log.Errorf("Scan result does not hold the image identifier. Error: %s", result.Error)
+					continue
+				}
+				status := workloadmeta.Success
+				reportedError := ""
+				var report *cyclonedx.BOM
+				if result.Error != nil {
+					// TODO: add a retry mechanism for retryable errors
+					log.Errorf("Failed to generate SBOM for docker: %s", result.Error)
 					status = workloadmeta.Failed
 					reportedError = result.Error.Error()
+				} else {
+					bom, err := result.Report.ToCycloneDX()
+					if err != nil {
+						log.Errorf("Failed to extract SBOM from report")
+						status = workloadmeta.Failed
+						reportedError = result.Error.Error()
+					}
+					report = bom
```

Comment:
for clarity this part can be extracted in a function.

Reply:
same for the previous `for loop` line 64 https://github.com/DataDog/datadog-agent/pull/22639/files#diff-84880b15ffd9ed1c97f5a97eed3b3bc24504f53ea426125b6925e89ecc0c6f8eR64